api/src/main/java/com/debugmafia/clueless/state/Game.java

```diff
@@ -0,0 +1,144 @@
+package com.debugmafia.clueless.state;
+
+import com.debugmafia.clueless.actions.*;
+import com.debugmafia.clueless.model.*;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.*;
+
+
+import javax.tools.DocumentationTool.Location;
+
+public class Game {
+    private HashSet<Card> winningCards;
+    private Board board;
+    private Turn turnInstance;
+    private List<Player> playerList;
+
+    public Game(Set<Player> players) {
+        this.board = new Board();
+        this.playerList = new ArrayList<>(players);
+        //TODO: Conversion above needs to have Mrs. Scarlett first
+        //TODO: Set the initialPlayers private List to the players
+        //TODO: Set the game state to GameState.IN_PROGRESS
+
+        this.winningCards = new HashSet<>(this.board.drawWinningCards());
+        //TODO: Determine how many cards go to each player (the number will always be a whole number),
+        //then call draw() for each player, giving them the cards that were drawn
+        
+        //TODO: Set the current players turn to the player who will go first, as well as other relevant turn
+        //information
+    }
+
+
+    public Game makeMove(Move m) {
+        //Restrictions: A piece cannot be moved to a location that does not exist, or to a HALLWAY location that
+        //already has a piece in it. Also, the player requesting to move must be equal to the current player.
+        this.board.movePiece(m.getPiece(), m.getTo());
+        //TODO: Update the turn state for the current player
+            // a. Add the move they just made
+            // b. Remove MOVE as an available action for their turn (meaning they cannot make
+            // another move)
+        return this;
+    }
+
+    public Game makeSuggestion(Suggestion s) {
+        //Restrictions: The player making the suggestion must also be the one who’s current turn it is. Also, the
+        //Location part of the suggestion must equal the location of the current players piece.
+        Card suggestedWeapon = this.board.getAssociatedCard(s.getWeapon());
+        Card suggestedRoom = this.board.getAssociatedCard(s.getRoom());
+        Card suggestedPiece = this.board.getAssociatedCard(s.getPiece());
+        //Can we make these into a Set for a suggestionSet?
+        
+        this.board.moveWeapon(s.getWeapon(), s.getRoom());
+        this.board.movePiece(s.getPiece(), s.getRoom());
+        
+
+        List<Player> validPlayers = this.playerList.stream().
+        filter(p -> p.hasCard(suggestedWeapon) || p.hasCard(suggestedRoom) || p.hasCard(suggestedPiece)).collect(Collectors.toList());
```

Comment:
Going off of my other comment regarding the for-loop, I think we can use the `findFirst()` stream method? This would also allow you to use the `isPresent()` method in your if statement below. So the change would look something like:

```java
Optional<Player> playerToRebut = this.playerList.stream().filter(/* query */).findFirst()

if (playerToRebut.isPresent()) {
    // do something with playerToRebut.get()
}
```

Reply:
Certainly. If we don't have to loop, Optional makes the most sense.